EugeneL97/DataStructuresAndAlgorithms
# Add Dynamic Array Implementation and Basic Tests

## Description


This PR introduces:
- A `DynamicArray` class that provides basic operations, including append, removeAt, setAt, and automatic resizing.
- Basic tests using Google Test to validate core functionality and edge-case handling for the `DynamicArray` class.

## Type of Changes


## Checklist of Changes

- [x] `Array/DynamicArray.h` – Implemented the `DynamicArray` template class.
- [x] `tests/test_dynamic_array.cpp` – Added tests to validate basic operations (append, removeAt, and edge cases).
- [x] `CMakeLists.txt` – Added GoogleTest and linked the `test_dynamic_array` executable.


## How to Test

1. Build the project using CMake.
2. Run the `test_dynamic_array` executable:
3. Verify the output:
```bash
[==========] 7 tests from 1 test suite ran. (0 ms total)
[  PASSED  ] 7 tests.
```
## Related Issues
N/A

## Screenshots (if applicable)
N/A

## Notes / Additional Context


- This PR is the foundation for future data structures, with each new structure following a similar pattern (header, implementation, and tests). I will be testing my knowledge and ability to adhere to industry standards.
- This project can be built and tested with any IDE that supports CMake or from the command line.
